jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -312,6 +313,95 @@ ecma_builtin_typedarray_prototype_for_each (ecma_value_t this_arg, /**< this arg
                                                          TYPEDARRAY_ROUTINE_FOREACH);
 } /* ecma_builtin_typedarray_prototype_for_each */
 
+
+#if ENABLED (JERRY_ES2015_BUILTIN_ITERATOR)
+/**
+ * Helper function for typedArray.prototype object's {'keys', 'values', 'entries', '@@iterator'}
+ * routines common parts.
+ *
+ * See also:
+ *          ECMA-262 v6, 22.2.3.15
+ *          ECMA-262 v6, 22.2.3.29
+ *          ECMA-262 v6, 22.2.3.6
+ *          ECMA-262 v6, 22.1.3.30
+ *
+ * Note:
+ *      Returned value must be freed with ecma_free_value.
+ *
+ * @return iterator result object, if success
+ *         error - otherwise
+ */
+static ecma_value_t
+ecma_builtin_typedarray_iterators_helper (ecma_value_t this_arg, /**< this argument */
+                                          uint8_t type) /**< any combination of ecma_iterator_type_t bits */
+{
+  ecma_value_t this_obj = ecma_op_to_object (this_arg);
```

Comment:
There is a conversion here, shouldn't the `this_obj` be freed at the end??

Reply:
The `ecma_deref_object (obj_p);` call stands for freeing `this_obj`.

Reply:
This arg must be an object so the conversion to object is unnecessary.